Js exercises refactor

JoeShep opened the pull request:
This PR is to update/refactor the orientation exercises. But note that I have a number of TODOs in here. Please take a look and then we can discuss resolving them to make this a merge-worthy PR

JoeShep commented:
@stevebrownlee When you have time, the TODO items need a look/discussion

stevebrownlee commented:
@JoeShep I'll review tonight

stevebrownlee commented on README.md:
```diff
@@ -12,7 +12,7 @@ In order to become a profitable, and sustainable organization, we need you to bu
 While you are building out the Bangazon Platform, your leadership team will ensure that the following foundational skills are developed by each team, and by each individual on the team.
 
-![Learning objectives](./learning-objectives.png)
+TODO: Replace with new visualization or list of learning objectives
```
Has something changed in the learning objectives?

JoeShep replied:
The whole branch on SOLID Principles is not something we cover (other than single responsibility, lightly). And we don't expressly teach the pillars of OOP either. Students may learn some of these things as a result of the work we do, but they aren't "taught". Certain students would look at this image and assume it translates directly to the curriculum, which could lead to confusion/disappointment when they don't receive specific instruction on those objectives.

stevebrownlee replied:
We do teach OOP but I agree on the SOLID stuff. We can nix that

stevebrownlee commented on orientation/exercises/01_DICTIONARIES.md:
````diff
@@ -37,4 +37,15 @@ purchases = [ ( 'GE', 100, '10-sep-2001', 48 ),
 Create a second purchase summary that which accumulates total investment by ticker symbol. In the above sample data, there are two blocks of GE. These can easily be combined by creating a dict where the key is the ticker and the value is the list of blocks purchased. The program makes one pass through the data to create the dict. A pass through the dict can then create a report showing each ticke ...
+
+Example output:
+```py
````
Changed markdown parser from `py` to `sh`

JoeShep replied:
👍

stevebrownlee commented on orientation/exercises/04_TUPLES.md:
```diff
@@ -19,6 +19,7 @@ subl .
 
 1. Create a tuple named `zoo` that contains your favorite animals.
 1. Find one of your animals using the `.index(value)` method on the tuple.
+<!-- TODO: Confirm this is what we want to do here. Should this be `foo in tuple` instead of a for loop? -->
```
I'm fine with changing this to value lookup syntax

JoeShep replied:
👍

stevebrownlee commented on orientation/exercises/10_TEST_ANIMALS.md:
```diff
@@ -19,8 +19,9 @@ As a team, we'll be building unit test coverage for all the functionality of the
 
 Write test cases to verify the I/O of the following methods of `Animal` and `Dog`.
 
+<!-- TODO: IN my code, I never create an instance of Animal. Is it necessary? -->
```
Yes, because there are requirements below to write tests for the `Animal` instance.

JoeShep replied:
👍

stevebrownlee commented on orientation/exercises/10_TEST_ANIMALS.md:
```diff
@@ -19,8 +19,9 @@ As a team, we'll be building unit test coverage for all the functionality of the
+<!-- TODO: IN my code, I never create an instance of Animal. Is it necessary? -->
 1. In the test class' `setUpClass()` method, create an instance of `Animal` and `Dog`.
-1. Animal object has the correct `name` property.
+1. Animal object has the correct `name` property.  <!-- What is correct name property?-->
```
When the student creates an `Animal` instance, it should have a `name` property. Perhaps clearify that in the previous step?

JoeShep replied:
👍

stevebrownlee commented on orientation/exercises/12_BAG_OF_LOOT.md:
```diff
@@ -48,6 +48,7 @@ You have an acquaintance whose job is to, once a year, delivery presents to the
 
 ## Requirements
 
+<!-- TODO: Remove this? We're not TDDing anymore -->
```
Correct

JoeShep replied:
👍

stevebrownlee commented on orientation/exercises/13_SHARED_MEMORIES.md:
```diff
@@ -25,6 +25,7 @@ touch margaret.py
     1. `EOFError`
     1. `KeyError`
 
+TODO: We didn't use pickle/serialization in 25. Is this exercise something to consider dumping?
```
Get rid of it

JoeShep replied:
👍

JoeShep commented:
@stevebrownlee Addressed your feedback. Also reordered exercises to put testing toward the end. Leaving Bag o Loot for now, but love idea of refactoring into SQL/db exercise. That will be a great addition.

stevebrownlee commented:
:shipit:
